contracts/swappa/SwappaRouterV1.sol

```diff
@@ -36,6 +51,14 @@ contract SwappaRouterV1 {
 		require(
 			ERC20(path[0]).transferFrom(msg.sender, pairs[0], inputAmount),
 			"SwappaRouter: Initial transferFrom failed!");
+		{
+			// Perform dryrun of swaps to verify the final output
+			uint dryrunAmount = this.dryrunSwap(path, pairs, extras, inputAmount);
```

Comment:
i dont think we can fit another boolean arg into the stack, hence why I had to indent this block to save on stack variables

Yeah I think it will just have to exist as a separate `dryrun()` function and the caller will need to be a smart contract that knows how to use dryrun correctly.

I have to write a separate set of arb contract anyways to take advantage of the recently deployed ChiTokens anyways (https://explorer.celo.org/address/0x62A25e867f7853B362BBDA8c0F400466A8063880/transactions)

Reply:
i dont think we can fit another boolean arg into the stack, hence why I had to indent this block to save on stack variables

Yeah I think it will just have to exist as a separate dryrun() function and the caller will need to be a smart contract that knows how to use dryrun correctly.

I have to write a separate set of arb contract anyways to take advantage of the recently deployed ChiTokens anyways (https://explorer.celo.org/address/0x62A25e867f7853B362BBDA8c0F400466A8063880/transactions)